modules/stochastic_tools/src/userobject/PolynomialChaos.C

```diff
@@ -189,6 +191,92 @@ PolynomialChaos::powerExpectation(const unsigned int n, const bool distributed)
   return val;
 }
 
+Real
+PolynomialChaos::computeDerivative(const unsigned int dim, const std::vector<Real> & x) const
+{
+  return computePartialDerivative({dim}, x);
+}
+
+Real
+PolynomialChaos::computePartialDerivative(const std::vector<unsigned int> & dim,
+                                          const std::vector<Real> & x) const
+{
+  mooseAssert(x.size() == _ndim, "Number of inputted parameters does not match PC model.");
+
+  std::vector<unsigned int> grad(_ndim);
+  for (auto d : dim)
+  {
+    mooseAssert(d < _ndim, "Specified dimension is greater than total number of parameters.");
+    grad[d]++;
+  }
+
+  DenseMatrix<Real> poly_val(_ndim, _order);
+
+  // Evaluate polynomials to avoid duplication
+  for (unsigned int d = 0; d < _ndim; ++d)
+    for (unsigned int i = 0; i < _order; ++i)
+      poly_val(d, i) = _poly[d]->computeDerivative(i, x[d], grad[d]);
+
+  Real val = 0;
+  for (unsigned int i = 0; i < _ncoeff; ++i)
```

Comment:
If you are looping over a std::vector you should use std::size_t, it is different than an unsigned int. You should probably use std::size_t for all indices/sizes that are used to access a std::vector container or any std container.

This will allow the code to work if we change to 64 bit integers and store huge amount of data, probably not the case for this stuff, but it is nice to get into the practice.

```suggestion
  for (std::size_t i = 0; i < _ncoeff; ++i)
```